src/event/spec.ts

```diff
@@ -3,28 +3,17 @@
  SPDX-License-Identifier: Apache-2.0
 */
 
-import Ajv, { Options } from "ajv";
 import { ValidationError } from "./validation";
 
 import { CloudEventV1 } from "./interfaces";
-import { schemaV1 } from "./schemas";
 import { Version } from "./cloudevent";
+import validate from "../schema/v1";
 
-const ajv = new Ajv({ extendRefs: true } as Options);
-
-// handle date-time format specially because a user could pass
-// Date().toString(), which is not spec compliant date-time format
-ajv.addFormat("js-date-time", function (dateTimeString) {
-  const date = new Date(Date.parse(dateTimeString));
-  return date.toString() !== "Invalid Date";
-});
-
-const isValidAgainstSchemaV1 = ajv.compile(schemaV1);
 
 export function validateCloudEvent<T>(event: CloudEventV1<T>): boolean {
   if (event.specversion === Version.V1) {
-    if (!isValidAgainstSchemaV1(event)) {
-      throw new ValidationError("invalid payload", isValidAgainstSchemaV1.errors);
+    if (!validate(event)) {
+      throw new ValidationError("invalid payload", (validate as any).errors);
```

Comment:
`(validate as any)` seems a bit off. Why do we need to cast to `any`?

I'm seeing `validate` as a function:

```ts
function validate(data: any, { instancePath, parentData, parentDataProperty, rootData }?: {
    instancePath?: string | undefined;
    parentData: any;
    parentDataProperty: any;
    rootData?: any;
}): boolean
```

Reply:
The function that AJV generates has an `errors` object attached to it if there are errors found when validating with `validate(data)`. It's a non-optimal API. Here you can see it in the generated validation code. I welcome suggestions on how to deal with this another way.

```js
function validate20(data, { instancePath = "", parentData, parentDataProperty, rootData = data } = {}) {
  let vErrors = null;
  let errors = 0;
  if (errors === 0) {
    if (data && typeof data == "object" && !Array.isArray(data)) {
      let missing0;
      if (
        (data.id === undefined && (missing0 = "id")) ||
        (data.source === undefined && (missing0 = "source")) ||
        (data.specversion === undefined && (missing0 = "specversion")) ||
        (data.type === undefined && (missing0 = "type"))
      ) {
        validate20.errors = [
          {
            instancePath,
            schemaPath: "#/required",
            keyword: "required",
            params: { missingProperty: missing0 },
            message: `must have required property '${  missing0  }'`,
          },
        ];
        return false;
      }
```